source/jupyter/wire/kernel.d

```diff
@@ -167,11 +175,77 @@ struct Kernel(Backend) if(isBackend!Backend) {
             version(JupyterLogVerbose) log("Told by the FE to execute code");
             handleExecuteRequest(requestMessage);
             return;
+
+        case "comm_open":
+            version(JupyterLogVerbose) log("Told by the FE to open a comm");
+            handleCommOpen(requestMessage);
+            return;
+
+        case "comm_msg":
+            version(JupyterLogVerbose) log("Received a comm msg from the FE");
+            handleCommMessage(requestMessage);
+            return;
+
+        case "comm_close":
+            version(JupyterLogVerbose) log("Told by the FE to close a comm");
+            handleCommClose(requestMessage);
+            return;
         }
 
         assert(0);
     }
 
+    void handleCommOpen(Message requestMessage) {
+        import jupyter.wire.message: commCloseMessage;
+        import std.traits: hasMember;
+
+        bool success = false;
+        scope(exit) if (!success) {
```

Comment:
Why not throw an exception and use `scope(failure)` instead?

Reply:
I suppose. It does make the control flow more readable. But I also need to send the comm_close message when the backend doesn't have comm_open implemented. see: https://jupyter-client.readthedocs.io/en/latest/messaging.html#opening-a-comm

I'll try to improve it.

Reply:
I ended up putting a try catch around the call to the backend. It might throw and in that case it is still good practice to send a comm_close message.

I didn't want the backend to signal not being able to handle a comm_open message by throwing (people can forget), so I kept the boolean.